src/httpServer/server.ts

```diff
@@ -423,6 +425,117 @@ export class DBOSHttpServer {
     dbosExec.logger.debug(`DBOS Server Registered List Workflow steps Get ${workflowStepsUrl}`);
   }
 
+  /**
+   *
+   * Register List Workflows endpoint.
+   * List workflows with optional filtering via request body.
+   */
+  static registerListWorkflowsEndpoint(dbosExec: DBOSExecutor, router: Router) {
+    const listWorkflowsUrl = '/workflows';
+    const listWorkflowsHandler = async (koaCtxt: Koa.Context) => {
+      const body = koaCtxt.request.body as {
+        workflow_ids?: string[];
+        workflow_name?: string;
+        authenticated_user?: string;
+        start_time?: string;
+        end_time?: string;
+        status?: string;
+        application_version?: string;
+        limit?: number;
+        offset?: number;
+        sort_desc?: boolean;
+        workflow_id_prefix?: string;
+      };
+
+      // Map request body keys to GetWorkflowsInput properties
+      const input: GetWorkflowsInput = {
+        workflowIDs: body.workflow_ids,
+        workflowName: body.workflow_name,
+        authenticatedUser: body.authenticated_user,
+        startTime: body.start_time,
+        endTime: body.end_time,
+        status: body.status as GetWorkflowsInput['status'],
+        applicationVersion: body.application_version,
+        limit: body.limit,
+        offset: body.offset,
+        sortDesc: body.sort_desc,
+        workflow_id_prefix: body.workflow_id_prefix,
+      };
+
+      const workflows = await dbosExec.listWorkflows(input);
+
+      // Map result to the underscore format.
+      koaCtxt.body = workflows.map((wf) => ({
+        workflow_id: wf.workflowID,
+        status: wf.status,
+        workflow_name: wf.workflowName,
+        workflow_class_name: wf.workflowClassName,
+        workflow_config_name: wf.workflowConfigName,
+        queue_name: wf.queueName,
+        authenticated_user: wf.authenticatedUser,
+        assumed_role: wf.assumedRole,
+        authenticated_roles: wf.authenticatedRoles,
+        output: wf.output,
+        error: wf.error,
+        input: wf.input,
+        executor_id: wf.executorId,
+        application_version: wf.applicationVersion,
+        application_id: wf.applicationID,
+        recovery_attempts: wf.recoveryAttempts,
+        created_at: wf.createdAt,
+        updated_at: wf.updatedAt,
+        timeout_ms: wf.timeoutMS,
+        deadline_epoch_ms: wf.deadlineEpochMS,
+      }));
```

Comment:
This is because the API must return the same format across languages, and we adopted the Python casing for other endpoints (except the predating queue metadata which is the other way around, and unfortunately wasn't considered when the workflow action endpoints where implemented)